---
title: "Figure4"
author: "Yayu WANG"
date: "2021/11/4"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

####Transcriptomic data analysis###

library(DESeq)
#### counts #######
data <- read.table("reads.counts.txt",sep="\t",header = T,row.names = 1)
database <- as.matrix(data)

######## 173 vs H2O #######
type <- factor(c("H2O","Y173"))
data.H2O.Y173 <- database[,c("H2O","Y173")]
data.H2O.Y173 <- data.H2O.Y173[rowSums(data.H2O.Y173)>0,]
cds <- newCountDataSet(data.H2O.Y173,type)
cds = estimateSizeFactors(cds)
cds <- estimateDispersions(cds, method="blind", sharingMode="fit-only" )
res = nbinomTest(cds, "H2O", "Y173" )
res2 <- res[abs(res$log2FoldChange)>1,]
res3 <- res2[res2$pval<0.05,]
write.table(res3,"H2O-VS-Y173.gene.diff2.exp.xls",sep="\t",row.names = F)

##### 2#######
type <- factor(c("H2O","Y594"))
data.H2O.Y594 <- database[,c("H2O","Y594")]
data.H2O.Y594 <- data.H2O.Y594[rowSums(data.H2O.Y594)>0,]
cds <- newCountDataSet(data.H2O.Y594,type)
cds = estimateSizeFactors(cds)
cds <- estimateDispersions(cds, method="blind", sharingMode="fit-only" )
res = nbinomTest(cds, "H2O", "Y594")
res2 <- res[abs(res$log2FoldChange)>1,]
res3 <- res2[res2$pval<0.05,]
write.table(res3,"H2O-VS-Y594.gene.diff2.exp.xls",sep="\t",row.names = F)

###### 3 #######
type <- factor(c("H2O","Y299"))
data.H2O.Y299 <- database[,c("H2O","Y299")]
data.H2O.Y299 <- data.H2O.Y299[rowSums(data.H2O.Y299)>0,]
cds <- newCountDataSet(data.H2O.Y299,type)
cds = estimateSizeFactors(cds)
cds <- estimateDispersions(cds, method="blind", sharingMode="fit-only" )
res = nbinomTest(cds, "H2O", "Y299" )
res2 <- res[abs(res$log2FoldChange)>1,]
res3 <- res2[res2$pval<0.05,]
write.table(res3,"H2O-VS-Y299.gene.diff2.exp.xls",sep="\t",row.names = F)

#####4 ####
type <- factor(c("H2O","Y550"))
data.H2O.Y550 <- database[,c("H2O","Y550")]
data.H2O.Y550 <- data.H2O.Y550[rowSums(data.H2O.Y550)>0,]
cds <- newCountDataSet(data.H2O.Y550,type)
cds = estimateSizeFactors(cds)
cds <- estimateDispersions(cds, method="blind", sharingMode="fit-only" )
res = nbinomTest(cds, "H2O", "Y550" )
res2 <- res[abs(res$log2FoldChange)>1,]
res3 <- res2[res2$pval<0.05,]
write.table(res3,"H2O-VS-Y550.gene.diff2.exp.xls",sep="\t",row.names = F)



####Figure4A####
library(ggplot2)
pathway=read.csv("pathway_enriched.csv",header=TRUE)
pathway$Pathway<-factor(pathway$Pathway,levels = unique(pathway$Pathway))
pathway$Strain=factor(pathway$Strain, levels = c("Strain 173", "Strain 594","Strain 299", "Strain 550"))

ggplot(pathway,aes(Qvalue,Pathway))+geom_point(aes(size=pathway$Genenumber,colour=pathway$Qvalue))+theme(axis.text=element_text(color='black'),axis.text.y=element_text(size=12),axis.text.x=element_text(size=9),panel.background=element_rect(fill='transparent'),panel.grid=element_line(color='grey'),panel.border=element_rect(fill='transparent',color='black'),axis.title=element_text(size=12))+scale_size_continuous(range=c(4,10))+scale_colour_gradient(low="red",  high="black") +labs(color="Qvalue",size="Gene number",x="Qvalue",y="Pathway name")+facet_grid(~Strain)+geom_vline(xintercept = 0.1,colour="grey")

####Figure4B####

library(VennDiagram)
upgene=read.csv("upgene_only.csv", header=TRUE, fill=T, na.string="", stringsAsFactors=F)

for (i in 1:4){
  tem = na.omit(upgene[,i])
  tem = data.frame(tem)
  assign(paste("a", i, sep = ""),tem)
}
venn_list <- list(a1[,1], a2[,1],a3[,1],a4[,1])

names(venn_list) <- colnames(upgene)

venn.diagram(venn_list, filename = 'venn.png', fill = c('red', 'purple', 'yellow', "green"))

####Figure4C#####
library(pheatmap)
upgene_KO=read.csv("upgene_ko.csv", header=TRUE, row.name=1)
P=pheatmap(log((use3+1),10),cellwidth = 15, scale="row", cellheight = 12,fontsize = 12, cluster_cols =FALSE,color =colorRampPalette(c("grey","orange","blue"))(10))
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r cars}
summary(cars)
```

## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
plot(pressure)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
